### ISSUE TITLE
Attach action for MorphToMany Relations is broken

### ISSUE THREAD

**Issue by captnCC** [laravel-json-api/laravel#102]
In the dummy app the attach action for MorphToMany relations is broken and throws a `LogicException`: `No schema for JSON:API resource type media.`

## Reproduction
I built a sample Laravel app from the included dummy:
https://github.com/byte-it/laravel-jsonapi-sample

A call  ` POST api/v1/posts/{post}/relations/media` then throws the mentioned error.

## Cause

The `LaravelJsonApi\Laravel\Routing\Route` class tries to retrieve the `Schema` for the related model. Because the MorphToMany relation returns it "morph" name instead of resolving to a real schema name, the resolving fails.
Digging deeper it becomes apparent, that it start in the `CollectionQuery` that tries to resolve the rules for the filters etc.
I don't see the reason at this point why these are needed at this point because the `POST` request doesn't support any query parameters anyways.

**Comment by lindyhopchris** [comment]
So the `POST` and `PATCH` request do support query parameters, which is why they are being validated.

As far as I'm aware this does work because it's passing in the tests for this package. Can I have a full stack trace for that error so I can see where it is coming from? And can you detail the request that is failing? That'll help me debug.

**Comment by captnCC** [comment]
The request: 
`POST /api/v1/posts/dWNDq/relationships/media`
```json
{
  "data": [
    {
      "type": "videos",
      "id": "0341be82-223f-4063-9b19-2b3516730efe"
    }
  ]
}
```
The same request as `PATCH` just works fine!

The complete trace:
```
[2021-07-05 13:44:12] local.ERROR: No schema for JSON:API resource type media. {"exception":"[object] (LogicException(code: 0): No schema for JSON:API resource type media. at json-api-test/vendor/laravel-json-api/core/src/Core/Schema/Container.php:106)
[stacktrace]
#0 json-api-test/vendor/laravel-json-api/laravel/src/Routing/Route.php(200): LaravelJsonApi\\Core\\Schema\\Container->schemaFor('media')
#1 json-api-test/vendor/laravel-json-api/validation/src/Rule.php(138): LaravelJsonApi\\Laravel\\Routing\\Route->inverse()
#2 json-api-test/vendor/laravel-json-api/laravel/src/Http/Requests/AnonymousCollectionQuery.php(42): LaravelJsonApi\\Validation\\Rule::filter()
#3 json-api-test/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): LaravelJsonApi\\Laravel\\Http\\Requests\\AnonymousCollectionQuery->rules()
#4 json-api-test/vendor/laravel/framework/src/Illuminate/Container/Util.php(40): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#5 json-api-test/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure(Object(Closure))
#6 json-api-test/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\\Container\\BoundMethod::callBoundMethod(Object(Illuminate\\Foundation\\Application), Array, Object(Closure))
#7 json-api-test/vendor/laravel/framework/src/Illuminate/Container/Container.php(651): Illuminate\\Container\\BoundMethod::call(Object(Illuminate\\Foundation\\Application), Array, Array, NULL)
#8 json-api-test/vendor/laravel/framework/src/Illuminate/Foundation/Http/FormRequest.php(112): Illuminate\\Container\\Container->call(Array)
#9 json-api-test/vendor/laravel/framework/src/Illuminate/Foundation/Http/FormRequest.php(91): Illuminate\\Foundation\\Http\\FormRequest->createDefaultValidator(Object(Illuminate\\Validation\\Factory))
#10 json-api-test/vendor/laravel/framework/src/Illuminate/Validation/ValidatesWhenResolvedTrait.php(23): Illuminate\\Foundation\\Http\\FormRequest->getValidatorInstance()
#11 json-api-test/vendor/laravel/framework/src/Illuminate/Foundation/Providers/FormRequestServiceProvider.php(30): Illuminate\\Foundation\\Http\\FormRequest->validateResolved()
#12 json-api-test/vendor/laravel/framework/src/Illuminate/Container/Container.php(1266): Illuminate\\Foundation\\Providers\\FormRequestServiceProvider->Illuminate\\Foundation\\Providers\\{closure}(Object(LaravelJsonApi\\Laravel\\Http\\Requests\\AnonymousCollectionQuery), Object(Illuminate\\Foundation\\Application))
#13 json-api-test/vendor/laravel/framework/src/Illuminate/Container/Container.php(1230): Illuminate\\Container\\Container->fireCallbackArray(Object(LaravelJsonApi\\Laravel\\Http\\Requests\\AnonymousCollectionQuery), Array)
#14 json-api-test/vendor/laravel/framework/src/Illuminate/Container/Container.php(1215): Illuminate\\Container\\Container->fireAfterResolvingCallbacks('LaravelJsonApi\\\\...', Object(LaravelJsonApi\\Laravel\\Http\\Requests\\AnonymousCollectionQuery))
#15 json-api-test/vendor/laravel/framework/src/Illuminate/Container/Container.php(774): Illuminate\\Container\\Container->fireResolvingCallbacks('LaravelJsonApi\\\\...', Object(LaravelJsonApi\\Laravel\\Http\\Requests\\AnonymousCollectionQuery))
#16 json-api-test/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(841): Illuminate\\Container\\Container->resolve('LaravelJsonApi\\\\...', Array, true)
#17 json-api-test/vendor/laravel/framework/src/Illuminate/Container/Container.php(692): Illuminate\\Foundation\\Application->resolve('LaravelJsonApi\\\\...', Array)
#18 json-api-test/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(826): Illuminate\\Container\\Container->make('LaravelJsonApi\\\\...', Array)
#19 json-api-test/vendor/laravel-json-api/laravel/src/Http/Requests/RequestResolver.php(111): Illuminate\\Foundation\\Application->make('LaravelJsonApi\\\\...')
#20 json-api-test/vendor/laravel-json-api/laravel/src/Http/Requests/ResourceQuery.php(83): LaravelJsonApi\\Laravel\\Http\\Requests\\RequestResolver->__invoke('posts')
#21 json-api-test/vendor/laravel-json-api/laravel/src/Http/Controllers/Actions/AttachRelationship.php(55): LaravelJsonApi\\Laravel\\Http\\Requests\\ResourceQuery::queryMany('posts')
#22 json-api-test/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(54): App\\Http\\Controllers\\Api\\V1\\PostController->attachRelationship(Object(LaravelJsonApi\\Laravel\\Routing\\Route), Object(LaravelJsonApi\\Core\\Store\\Store), Object(App\\Models\\Post), 'posts', 'post', 'media')
#23 json-api-test/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(45): Illuminate\\Routing\\Controller->callAction('attachRelations...', Array)
#24 json-api-test/vendor/laravel/framework/src/Illuminate/Routing/Route.php(254): Illuminate\\Routing\\ControllerDispatcher->dispatch(Object(Illuminate\\Routing\\Route), Object(App\\Http\\Controllers\\Api\\V1\\PostController), 'attachRelations...')
#25 json-api-test/vendor/laravel/framework/src/Illuminate/Routing/Route.php(197): Illuminate\\Routing\\Route->runController()
#26 json-api-test/vendor/laravel/framework/src/Illuminate/Routing/Router.php(695): Illuminate\\Routing\\Route->run()
#27 json-api-test/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#28 json-api-test/vendor/laravel-json-api/laravel/src/Http/Middleware/BootJsonApi.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#29 json-api-test/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): LaravelJsonApi\\Laravel\\Http\\Middleware\\BootJsonApi->handle(Object(Illuminate\\Http\\Request), Object(Closure), 'v1')
#30 json-api-test/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php(50): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#31 json-api-test/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Routing\\Middleware\\SubstituteBindings->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#32 json-api-test/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#33 json-api-test/vendor/laravel/framework/src/Illuminate/Routing/Router.php(697): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#34 json-api-test/vendor/laravel/framework/src/Illuminate/Routing/Router.php(672): Illuminate\\Routing\\Router->runRouteWithinStack(Object(Illuminate\\Routing\\Route), Object(Illuminate\\Http\\Request))
#35 json-api-test/vendor/laravel/framework/src/Illuminate/Routing/Router.php(636): Illuminate\\Routing\\Router->runRoute(Object(Illuminate\\Http\\Request), Object(Illuminate\\Routing\\Route))
#36 json-api-test/vendor/laravel/framework/src/Illuminate/Routing/Router.php(625): Illuminate\\Routing\\Router->dispatchToRoute(Object(Illuminate\\Http\\Request))
#37 json-api-test/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(166): Illuminate\\Routing\\Router->dispatch(Object(Illuminate\\Http\\Request))
#38 json-api-test/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}(Object(Illuminate\\Http\\Request))
#39 json-api-test/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#40 json-api-test/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php(31): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#41 json-api-test/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsToNull->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#42 json-api-test/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#43 json-api-test/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php(40): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#44 json-api-test/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\TrimStrings->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#45 json-api-test/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#46 json-api-test/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#47 json-api-test/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(86): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#48 json-api-test/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#49 json-api-test/vendor/fruitcake/laravel-cors/src/HandleCors.php(52): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#50 json-api-test/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fruitcake\\Cors\\HandleCors->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#51 json-api-test/vendor/fideloper/proxy/src/TrustProxies.php(57): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#52 json-api-test/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fideloper\\Proxy\\TrustProxies->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#53 json-api-test/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#54 json-api-test/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(141): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#55 json-api-test/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(110): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter(Object(Illuminate\\Http\\Request))
#56 json-api-test/public/index.php(52): Illuminate\\Foundation\\Http\\Kernel->handle(Object(Illuminate\\Http\\Request))
#57 {main}
"} 
```

**Comment by lindyhopchris** [comment]
Confirmed this is a bug. There are tests for the `PATCH` route, but not the `POST` route. Adding them fails.

**Comment by lindyhopchris** [comment]
Is a bug in the controller action - it's resolving the collection query using the resource type, not the relation's inverse resource type. Looks like the same bug is in the detach relationship action too, so need to add tests for that.

**Comment by lindyhopchris** [comment]
Fix will be in `1.0.0-beta.5`
